model/lifecycle.go

```diff
@@ -1442,14 +1442,26 @@ func AddNewTasks(ctx context.Context, activated bool, v *Version, p *Project, pa
 		// a record in the TVPairSet indicating that it should exist
 		tasksToAdd := []string{}
 		for _, taskname := range pairs.ExecTasks.TaskNames(b.BuildVariant) {
-			if _, ok := existingTasksIndex[taskname]; ok {
+			if t, ok := existingTasksIndex[taskname]; ok {
+				if !t.Activated && activated {
+					// update task activation
+					if _, err = t.ActivateTask(evergreen.User); err != nil {
+						return nil, errors.Wrapf(err, "problem updating active state for existing task '%s'", taskname)
+					}
+				}
 				continue
 			}
 			tasksToAdd = append(tasksToAdd, taskname)
 		}
 		displayTasksToAdd := []string{}
 		for _, taskname := range pairs.DisplayTasks.TaskNames(b.BuildVariant) {
-			if _, ok := existingTasksIndex[taskname]; ok {
+			if t, ok := existingTasksIndex[taskname]; ok {
```

Comment:
is this necessary? (can a task depend on a display task?)

Reply:
I don't actually think so, but I can't find proof of this so I figured better safe than sorry? But if you also feel confident then I can get rid of this.

Reply:
I'm fairly confident: I'm not sure what it would mean to depend on a display task.

Reply:
Found it: we only look in the [execTable](https://github.com/evergreen-ci/evergreen/blob/1c87d4fff7405b0d35b8ad416a5cc7a12b70ad0b/model/lifecycle.go#L882) for the dependency.
We should probably add it to the [validator](https://github.com/evergreen-ci/evergreen/blob/1c87d4fff7405b0d35b8ad416a5cc7a12b70ad0b/validator/project_validator.go#L979) at some point.